.github/workflows/e2e.yml

```diff
@@ -0,0 +1,76 @@
+##
+# Run e2e tests
+##
+name: e2e
+on:
+  pull_request:
+
+jobs:
+  pre_build:
```

Comment:
Can we call this `setup`? I guess it technically precedes the `build` step, but that step both builds and tests, and it's really about getting us to the test.

Reply:
Have renamed